{"url": "https://api.github.com/repos/datalad/datalad/issues/5272", "repository_url": "https://api.github.com/repos/datalad/datalad", "labels_url": "https://api.github.com/repos/datalad/datalad/issues/5272/labels{/name}", "comments_url": "https://api.github.com/repos/datalad/datalad/issues/5272/comments", "events_url": "https://api.github.com/repos/datalad/datalad/issues/5272/events", "html_url": "https://github.com/datalad/datalad/issues/5272", "id": 777731990, "node_id": "MDU6SXNzdWU3Nzc3MzE5OTA=", "number": 5272, "title": "AnnexRepo.add_submodule: unused and broken for adjusted mode repos", "user": {"login": "yarikoptic", "id": 39889, "node_id": "MDQ6VXNlcjM5ODg5", "avatar_url": "https://avatars.githubusercontent.com/u/39889?v=4", "gravatar_id": "", "url": "https://api.github.com/users/yarikoptic", "html_url": "https://github.com/yarikoptic", "followers_url": "https://api.github.com/users/yarikoptic/followers", "following_url": "https://api.github.com/users/yarikoptic/following{/other_user}", "gists_url": "https://api.github.com/users/yarikoptic/gists{/gist_id}", "starred_url": "https://api.github.com/users/yarikoptic/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/yarikoptic/subscriptions", "organizations_url": "https://api.github.com/users/yarikoptic/orgs", "repos_url": "https://api.github.com/users/yarikoptic/repos", "events_url": "https://api.github.com/users/yarikoptic/events{/privacy}", "received_events_url": "https://api.github.com/users/yarikoptic/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 63778595, "node_id": "MDU6TGFiZWw2Mzc3ODU5NQ==", "url": "https://api.github.com/repos/datalad/datalad/labels/wontfix", "name": "wontfix", "color": "ffffff", "default": true, "description": "Issue is unlikely to be addressed"}, {"id": 1757223715, "node_id": "MDU6TGFiZWwxNzU3MjIzNzE1", "url": "https://api.github.com/repos/datalad/datalad/labels/adjusted-branches", "name": "adjusted-branches", "color": "0052cc", "default": false, "description": "Issues caused by or affected adjusted branch operation"}, {"id": 2096760260, "node_id": "MDU6TGFiZWwyMDk2NzYwMjYw", "url": "https://api.github.com/repos/datalad/datalad/labels/corpse-in-basement", "name": "corpse-in-basement", "color": "d4c5f9", "default": false, "description": ""}], "state": "open", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 3, "created_at": "2021-01-03T22:54:14Z", "updated_at": "2021-01-15T08:46:17Z", "closed_at": null, "author_association": "MEMBER", "active_lock_reason": null, "body": "https://github.com/datalad/git-annex/runs/1638513807?check_suite_focus=true#step:21:1469\r\n\r\n```\r\n======================================================================\r\nERROR: datalad.support.tests.test_annexrepo.test_AnnexRepo_add_submodule\r\n----------------------------------------------------------------------\r\nTraceback (most recent call last):\r\n  File \"C:\\hostedtoolcache\\windows\\Python\\3.7.9\\x64\\lib\\site-packages\\nose\\case.py\", line 198, in runTest\r\n    self.test(*self.arg)\r\n  File \"C:\\hostedtoolcache\\windows\\Python\\3.7.9\\x64\\lib\\site-packages\\datalad\\tests\\utils.py\", line 1092, in _wrap_with_testrepos\r\n    t(*(arg + (uri,)), **kw)\r\n  File \"C:\\hostedtoolcache\\windows\\Python\\3.7.9\\x64\\lib\\site-packages\\datalad\\tests\\utils.py\", line 757, in _wrap_with_tempfile\r\n    return t(*(arg + (filename,)), **kw)\r\n  File \"C:\\hostedtoolcache\\windows\\Python\\3.7.9\\x64\\lib\\site-packages\\datalad\\support\\tests\\test_annexrepo.py\", line 1756, in test_AnnexRepo_add_submodule\r\n    assert_repo_status(top_repo, annex=True)\r\n  File \"C:\\hostedtoolcache\\windows\\Python\\3.7.9\\x64\\lib\\site-packages\\datalad\\tests\\utils.py\", line 1664, in assert_repo_status\r\n    status = r.status(untracked=untracked_mode)\r\n  File \"C:\\hostedtoolcache\\windows\\Python\\3.7.9\\x64\\lib\\site-packages\\datalad\\support\\gitrepo.py\", line 3611, in status\r\n    eval_submodule_state=eval_submodule_state)\r\n  File \"C:\\hostedtoolcache\\windows\\Python\\3.7.9\\x64\\lib\\site-packages\\datalad\\support\\gitrepo.py\", line 3800, in diffstatus\r\n    subrepo_commit = subrepo.get_hexsha(subrepo.get_corresponding_branch())\r\n  File \"C:\\hostedtoolcache\\windows\\Python\\3.7.9\\x64\\lib\\site-packages\\datalad\\support\\gitrepo.py\", line 1856, in get_hexsha\r\n    raise ValueError(\"Unknown commit identifier: %s\" % commitish)\r\nValueError: Unknown commit identifier: dl-test-branch\r\n```", "closed_by": null, "reactions": {"url": "https://api.github.com/repos/datalad/datalad/issues/5272/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/datalad/datalad/issues/5272/timeline", "performed_via_github_app": null}